apps/contributions/tests/test_models.py

```diff
@@ -2088,6 +2088,21 @@ def test_unmarked_abandoned_carts(self, unmarked_abandoned_contributions):
             c.id for c in unmarked_abandoned_contributions
         }
 
+    def test_with_first_payment_date_when_payments_exist(self, monthly_contribution_multiple_payments):
+        contributions = Contribution.objects.with_first_payment_date()
+        first_payment = (
+            Payment.objects.filter(contribution_id=monthly_contribution_multiple_payments.id)
+            .order_by("transaction_time")
+            .first()
+        )
+        assert len(contributions) == 1
+        assert contributions[0].first_payment_date == first_payment.transaction_time
+
+    def test_with_first_payment_date_when_no_payments(self, one_time_contribution):
+        contributions = Contribution.objects.with_first_payment_date()
+        assert len(contributions) == 1
+        assert contributions[0].first_payment_date is None
+
```

Comment:
```suggestion
    def test_with_first_payment_date_when_no_payments(self, one_time_contribution):
        contributions = Contribution.objects.with_first_payment_date()
        assert contributions.count() == 1
        assert contributions.first().first_payment_date is None
```

Also, might be good to have an initial assertion that the contribution has no payments.

Reply:
👍 6e903fec3082f59f7f165f3436cd0c7c1aa6a7cb